docs: update vault-token note in job run command #8040 (#12385)
hashicorp/nomad: website/content/docs/commands/job/run.mdx

  a Consul token must be supplied with appropriate service and kv Consul ACL policy
  permissions.

- `-vault-token`: Used to validate if the user submitting the job has
  permission to run the job according to its Vault policies.

>   A Vault token must be supplied if `allow_unauthenticated = false` is

Reviewer: Maybe add a link to the `allow_unauthenticated` config page.

Author: added link, and had to modify another link since consul also has an `allow_unauthenticated` that's referenced right above here

  configured in the Vault server. If set, the passed Vault token is added to the jobspec
  before sending to the Nomad servers. This allows passing the vault token without
  storing it in the job file. This overrides the token found in the

>   `$VAULT_TOKEN` environment variable and the `vault-token` field in the job file.

Reviewer: ```suggestion
  `$VAULT_TOKEN` environment variable and the `vault_token` field in the job file.
```

Adding a link to `vault_token` would be helpful as well.

Author: linked

  This token is cleared from the job after validating and cannot be used within the

>   job executing environment. If templating a Vault token within the job is needed,
>   please use the `vault` stanza.

Reviewer: ```suggestion
  job executing environment. Use the `vault` stanza when templating in a job with a Vault token.
```
